### ISSUE TITLE
Indent "first" style should have a configurable indent level.

### ISSUE THREAD

**Issue by maxbogue** [eslint/eslint#10496]
**What rule do you want to change?**
`indent` options that can be set to `"first"`.

**Does this change cause the rule to produce more or fewer warnings?**
It would be an option, so neither.

**How will the change be implemented? (New option, new default behavior, etc.)?**
eslint assumes an indent level of 1 one when the first parameter/argument is already on its own line. I want to be able to configure that indent level. I'm not sure the best config format; maybe `["first", 2]` or an object like `{ first: true, indent: 2 }`.

**Please provide some example code that this change will affect:**

```js
thisIsALongFunctionName(thisIsALongParameterOne,
    thisIsALongParameterTwo);
thisIsALongFunctionName(
    thisIsALongParameterOne,
    thisIsALongParameterTwo)
```

**What does the rule currently do for this code?**
```js
thisIsALongFunctionName(thisIsALongParameterOne,
                        thisIsALongParameterTwo);
// Note these parameters are only indented 1 level.
thisIsALongFunctionName(
  thisIsALongParameterOne,
  thisIsALongParameterTwo)
```

**What will the rule do after it's changed?**
With a configuration of `["first", 2]` it would do this:
```js
thisIsALongFunctionName(thisIsALongParameterOne,
                        thisIsALongParameterTwo);
// Now these parameters are indented 2 levels.
thisIsALongFunctionName(
    thisIsALongParameterOne,
    thisIsALongParameterTwo)
```

**Comment by not-an-aardvark** [comment]
Hi, thanks for the proposal. I'm personally not sure it would be worthwhile to add this option because the `indent` rule is already extremely complex, and so I think we should only add additional options if they would benefit a large number of people. That said, I'm not opposed to this if other team members think it would be worthwhile, so let's see what others say.

**Comment by maxbogue** [comment]
If you don't want to add complexity, you could make "first" default to indent level 2, but that would be a more breaking change. I'm not sure why it defaults to 1; in my experience new blocks are indented 1 but line continuations are indented 2, and these are all line continuations.

**Comment by platinumazure** [comment]
I don't follow. I thought the "first" option meant that later arguments (elements, properties, etc.) would align with the first one, no matter how many indent levels you configure. I don't believe the "first" option would move a token from one line to the next (no indent option should do that).

Am I missing something? I don't know what the issue is here.

**Comment by not-an-aardvark** [comment]
The `"first"` option requires that all elements after the first element must be aligned with the first element. This issue is about how the first element itself should be indented in the event that it appears at the start of a line.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
